geometric/run_json.py

```diff
@@ -62,16 +62,28 @@ def get_output_json_dict(in_json_dict, schema_traj):
     })
     return out_json_dict
 
+def make_constraints_string(constraints_dict):
+    """ Convert the new constraints dict format into the original string format """
+    constraints_string = ''
+    for key, value_list in constraints_dict.items():
+        if key not in ('freeze', 'set', 'scan'):
+            print("Warning, constraints key %s is not recognized" % key)
+        constraints_string += '$' + key + '\n'
+        for spec_tuple in value_list:
+            constraints_string += ' '.join(spec_tuple) + '\n'
+    return constraints_string
+
 def geometric_run_json(in_json_dict):
     """ Take a input dictionary loaded from json, and return an output dictionary for json """
     input_opts = parse_input_json_dict(copy.deepcopy(in_json_dict))
     M, engine = geometric.optimize.get_molecule_engine(**input_opts)
     # Get initial coordinates in bohr
     coords = M.xyzs[0].flatten() * geometric.nifty.ang2bohr
     # Read in the constraints
-    constraints_string = input_opts.get('constraints', None)
+    constraints_dict = input_opts.get('constraints', {})
+    constraints_string = make_constraints_string(constraints_dict)
```

Comment:
A test should probably be added as this functionality is not currently tested anywhere. This will also break current Crank tests.

Reply:
Yes, I have prepared the update in crank tests and they will be fixed as soon as this PR is merged.